MBC7: Note about HALT and sensor relying on PHI/ cart clock (#596)
gbdev/pandocs: src/MBC7.md

(Partially unconfirmed). Note that you cannot re-latch the accelerometer
value without first erasing it; attempts to do so yield no change.

> Once the latch for a sample update has been triggered the cpu should not be

Reviewer: ```suggestion
Once the latch for a sample update has been triggered the CPU should not be
```

> put into HALT for at least 1.2 msec since HALTing turns off the PHI cartridge

Reviewer: ```suggestion
put into HALT for at least 1.2 µs since HALTing turns off the PHI cartridge
```

Reviewer 2: Note that we've been trying to phase out the “millisecond” or “microsecond” terminology from Pan Docs, since it's not clear how it relates to CPU cycles (the most practical unit, since it's how instructions are measured).

Unfortunately, the PR's already been merged, so that's too late. But I want to have this note out there for future reference.

(I am aware that the relationship *is* clear, in principle, but it's not clear to newbies. Possibly there should be some introductory page clarifying the various timing units. Idk.)

Reviewer 2: Using cycles makes sense for the CPU, but less so for other hardware (the PPU uses “dots”, the APU is talked about in “<var>n</var> single-speed cycles / 2<var>n</var> double-speed cycles”... so using milliseconds is, in principle, more sensible; however, we stumble upon the same 1000 / 1024 distinction as e.g. MB vs. MiB; and I don't know a sensible solution to this.

Reviewer: @bbbbbr yep, you're right, sorry I was too quick, I'll push the change now

Author: Keep in mind this is an external sensor on a cartridge- it's not uncommon for physical sensors to have settling times which are measured in absolute temporal units (often temperature dependent) that aren't entirely dependent on the clock frequency being provided to the sensor.

> Unfortunately, the PR's already been merged, so that's too late. But I want to have this note out there for future reference.

Given the above, it's not clear (without deeper research into the sensor part) whether it *would be* appropriate in the first place to measure this in GB CPU cycles or some derivative of those.

> so using milliseconds is, in principle, more sensible; however, we stumble upon the same 1000 / 1024 distinction as e.g. MB vs. MiB; and I don't know a sensible solution to this.

I don't understand how the 1000 vs 1024 trickiness would apply to time units.

Reviewer: @ISSOtm The purpose of this PR was to highlight the potential inaccuracy of the values retrieved from the accelerometer in a precise situation, not to initiate a migration to a different unit for time.

The whole document is full of such terminology so if you want to initiate such document-wide change, we should first set the 'reference' in the CONTRIBUTING.md's 'Units and prefixes' guideline chapter (https://github.com/gbdev/pandocs/blob/master/CONTRIBUTING.md#2-units-and-prefixes ). Once that's settled and we have clear in mind when to use cycles vs 'normal' timing we can initiate a 'migration'.

Reviewer: @ISSOtm could you please follow this up with a PR to CONTRIBUTING.md and/or RFC 'Issue' about this change, if you'd like to pursue it ?

> signal (clock from the GB) which the MBC7 accelerometer sensor relies on.

Reviewer: ```suggestion
signal (the System Clock) which the MBC7 accelerometer sensor relies on.
```

Author: PHI clock runs at a different speed than the System Clock, so they are not interchangeable. It might be described as "a clock signal from the CPU" 

From elsewhere in the pandocs:
https://gbdev.io/pandocs/Gameboy_Camera.html
> The Game Boy Camera cartridge is one of the few cartridges that use the PHI signal (clock from the GB). That signal is a 1 MiHz clock (1047567 Hz). 

I guess that might mean this part of the Pandocs needs updating:
https://gbdev.io/pandocs/External_Connectors.html?highlight=PHI#cartridge-slot
> `2	PHI	System Clock`

Reviewer: Having it inconsistent with how PHI is described at its respective page would've been confusing. Let's update it at the PHI definitions and then propagate the change once that change is discussed/applied

Reviewer: Could you please file an Issue regarding this? @bbbbbr

Author: Sounds like a good idea. Probably good to confirm that PHI is 2MHz in GBC double speed mode (I see it implied in some places, but don't see it explicitly documented).

Author: >  Let's update it at the PHI definitions and then propagate the change once that change is discussed/applied

Looking over pandocs, maybe PHI could remain as System Clock if it's speed is indicated as 1MHz and the 4MHz clock has a distinct name as Master Clock or Oscillator Clock. Right now there is a mix of things in various places which for me, at least, leads to confusion.

Here's a PR that attempts to clarify that #597

> If that clock is turned off too soon the X and Y values will have signficantly

Reviewer: ```suggestion
If that clock is turned off too soon the X and Y values will have significantly
```
